pkg/storage/memory/memory.go

```diff
@@ -68,9 +91,19 @@ func (s *storageMem) ReadAndDestroy(id string) (string, error) {
 
 	defer delete(s.store, id)
 
-	if !secret.Expiry.IsZero() && secret.Expiry.Before(time.Now()) {
+	// Still check to see if the secret has expired in order to prevent a
+	// race condition where a secret has expired but the the store pruner has
+	// not yet been invoked.
+	if secret.hasExpired() {
 		return "", storage.ErrSecretNotFound
 	}
 
 	return secret.Secret, nil
 }
+
+func (m *memStorageSecret) hasExpired() bool {
+	if !m.Expiry.IsZero() && m.Expiry.Before(time.Now()) {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	return !m.Expiry.IsZero() && m.Expiry.Before(time.Now())
```

No need to `if` on the expression if its `if […] { return true } return false`

Reply:
Fair, Implemented.